crates/runtime-table-partition/src/insert.rs

```diff
@@ -405,7 +405,11 @@ impl DisplayAs for PartitionInputExec {
 /// Evaluate the `physical_expr` for each row in `batch`. A partition batch is
 /// created for each unique value produced by evaluating the expression
 /// containing the rows that produced that unique partition value.
-fn partition_batch(
+///
+/// # Errors
+/// Returns an error when the expressions cannot be evaluated, the batch cannot
+/// be partitioned, Arrays cannot be created or the batch cannot be filtered.
+pub fn partition_batch(
```

Comment:
Because we aren't reusing `PartitionedTableProvider` yet, this partitioning logic has to be exposed.